beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -96,22 +96,26 @@ func CanProcessValidatorRegistry(state *pb.BeaconState) bool {
 
 // ProcessEth1Data processes eth1 block deposit roots by checking its vote count.
 // With sufficient votes (>2*EPOCHS_PER_ETH1_VOTING_PERIOD), it then
-// marks the voted Eth1 data as the latest data set.
+// marks the block Eth1 data as the latest data set.
 //
 // Official spec definition:
-//     if eth1_data_vote.vote_count * 2 > EPOCHS_PER_ETH1_VOTING_PERIOD * SLOTS_PER_EPOCH for
-//       some eth1_data_vote in state.eth1_data_votes.
-//       (ie. more than half the votes in this voting period were for that value)
-//       Set state.latest_eth1_data = eth1_data_vote.eth1_data.
-//		 Set state.eth1_data_votes = [].
-//
-func ProcessEth1Data(state *pb.BeaconState) *pb.BeaconState {
+//def process_eth1_data(state: BeaconState, block: BeaconBlock) -> None:
+//    state.eth1_data_votes.append(block.body.eth1_data)
+//    if state.eth1_data_votes.count(block.body.eth1_data) * 2 > SLOTS_PER_ETH1_VOTING_PERIOD:
+//        state.latest_eth1_data = block.body.eth1_data
+func ProcessEth1Data(state *pb.BeaconState, block *pb.BeaconBlock) *pb.BeaconState {
+	eth1DataBlockVote := &pb.Eth1DataVote{Eth1Data: block.Eth1Data}
+	state.Eth1DataVotes = append(state.Eth1DataVotes, eth1DataBlockVote)
+	var count uint64
 	for _, eth1DataVote := range state.Eth1DataVotes {
-		if eth1DataVote.VoteCount*2 > params.BeaconConfig().SlotsPerEpoch*
-			params.BeaconConfig().EpochsPerEth1VotingPeriod {
-			state.LatestEth1Data = eth1DataVote.Eth1Data
+		if eth1DataVote.Eth1Data == block.Eth1Data {
+			count++
 		}
 	}
+	if count*2 > params.BeaconConfig().SlotsPerEpoch*
```

Comment:
SLOTS_PER_ETH1_VOTING_PERIOD should be a constant . its name is SlotsPerEth1VotingPeriod. please dont calculate it
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#time-parameters